diff -Nuar elfutils-0.166.orig/libdw/Makefile.in elfutils-0.166/libdw/Makefile.in
--- elfutils-0.166.orig/libdw/Makefile.in	2016-03-31 09:59:09.000000000 +0100
+++ elfutils-0.166/libdw/Makefile.in	2016-07-08 21:42:01.329455945 +0100
@@ -1003,7 +1003,7 @@
 		-Wl,--enable-new-dtags,-rpath,$(pkglibdir) \
 		-Wl,--version-script,$<,--no-undefined \
 		-Wl,--whole-archive $(filter-out $<,$^) -Wl,--no-whole-archive\
-		-ldl -lz $(argp_LDADD) $(zip_LIBS)
+		-ldl -lz -lfts $(argp_LDADD) $(zip_LIBS)
 	@$(textrel_check)
 	$(AM_V_at)ln -fs $@ $@.$(VERSION)
 
