diff -Nuar elfutils-0.166.orig/src/Makefile.am elfutils-0.166/src/Makefile.am
--- elfutils-0.166.orig/src/Makefile.am	2016-03-31 09:59:09.000000000 +0100
+++ elfutils-0.166/src/Makefile.am	2016-07-09 15:31:00.369375737 +0100
@@ -109,7 +109,7 @@
 	   $(demanglelib)
 size_LDADD = $(libelf) $(libeu) $(argp_LDADD)
 strip_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl
-ld_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl
+ld_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl -lobstack
 if NATIVE_LD
 # -ldl is always needed for libebl.
 ld_LDADD += libld_elf.a
diff -Nuar elfutils-0.166.orig/src/Makefile.in elfutils-0.166/src/Makefile.in
--- elfutils-0.166.orig/src/Makefile.in	2016-03-31 09:59:09.000000000 +0100
+++ elfutils-0.166/src/Makefile.in	2016-07-09 15:31:10.849453390 +0100
@@ -557,7 +557,7 @@
 
 size_LDADD = $(libelf) $(libeu) $(argp_LDADD)
 strip_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl
-ld_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl \
+ld_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl -lobstack \
 	$(am__append_3)
 ld_LDFLAGS = -rdynamic
 elflint_LDADD = $(libebl) $(libelf) $(libeu) $(argp_LDADD) -ldl
